ignite/templates/app/files/app/app.go.plush

```diff
@@ -220,8 +157,13 @@ func New(
 			depinject.Supply(
 				// supply the application options
 				appOpts,
-                // supply ibc keeper getter for the IBC modules
-                app.GetIBCeKeeper,
+                // Supply with IBC keeper getter for the IBC modules with App Wiring.
+                // The IBC Keeper cannot be passed because it has not been initiated yet.
+                // Passing the getter, the app IBC Keeper will always be accessible.
+                // This needs to be removed after IBC supported App Wiring.
+                app.GetIBCKeeper,
```

Comment:
I don't think this works as expected. Unlike `depinject.Provide()` which takes provider functions, `depinject.Supply()` takes direct values. 

So here the value is a function that returns the `IBCKeeper`, not the `IBCKeeper` itself. So the wire will work only if the dependency require a function that returns the `IBCKeeper` but not if the dependency requires a `IBCKeeper`.

I'm not 100% sure of this, but that's what I understood while I had to inject the `consumerKeeper` in place of the `stakingKeeper`. Maybe @julienrbrt can confirm/negate ?

Reply:
Yes, this is correct, and you can see back this on how `IBCKeeper` is used in IBC-enabled modules: https://github.com/ignite/cli/blob/7d54608/ignite/templates/module/create/files/base/x/%7B%7BmoduleName%7D%7D/keeper/keeper.go.plush#L42

Reply:
Oh right I missed the part where the `NewKeeper` takes the `ibcKeeperFn` as a parameter. All OK then! 
Thanks Julien for the explanation.